### PR TITLE
components: Promote Heading

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Promotes the `Heading` component to a full `components` export.

## How has this been tested?
Storybook still works. The export is experimental and unused so there's no where else to test it.

## Types of changes
New feature.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/components/src/ui/card/README.md [hunk 1]
```diff
@@ -13,7 +13,8 @@ import {
 	CardBody,
 	CardFooter,
 	Text,
-} from '@wordpress/components/ui';
+	Heading,
```
Comment (gziolo) [inline packages/components/src/ui/card/README.md:16]: It's exposed in documentation as `__experimentalHeading`, so I guess it should be aliased here.

File: packages/components/src/heading/index.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export { default } from './component';
+export * from './hook';
```
Comment (gziolo) [inline packages/components/src/heading/index.ts]: This catch-all export made me think. What do we really expose from there? It doesn't seem to be used elsewhere.

**Comment by gziolo** [comment]
I see a few PRs that expose G2 components as experimental. A few things to think about:
- when we move the folder out of the `ui` folder then the README files are going to be exposed to the developer portal so we need to put extra care that it’s usable
- how to ensure that those components are marked as experimental in Storybook
- do we need to keep them experimental, what do we worry about?
- should we expose also hooks that exist with the components?

**Comment by sarayourfriend** [comment]
> when we move the folder out of the ui folder then the README files are going to be exposed to the developer portal so we need to put extra care that it’s usable

👍 What did you have in mind here? Are there specific things we should be looking for?

> how to ensure that those components are marked as experimental in Storybook

We didn't previously call out experimental components as experimental in Storybook (see `Text` for example). Is this something new? I'd propose just adding a `Components (Experimental)/` namespace for Storybook.

> do we need to keep them experimental, what do we worry about?

Yes we should keep them experimental for now until we've migrated more of them and confirmed that the implementations and props are what we want. Heading is simple enough that I don't think it needs to be experimental but it implements all of Text which is experimental and wide ranging (has truncation, highlighting, etc). I think the components are relatively stable but it'd be good for more people to review them so we can continue iterating on them.

> should we expose also hooks that exist with the components?

I think the hooks are good for internal library composability but not necessary for things outside the library. What do you think? /cc @youknowriad

**Comment by youknowriad** [comment]
My thinking right now is that we should hold a little bit before making things stable (after feature freeze for WP 5.8 which is something like May 19th in the Gutenberg repo) and after that date, I believe we should be more liberal about stabilizing components (and ideally use them in our codebase) cause we'll have time to adapt in case we discover anything weird (until WP 5.9)

**Comment by gziolo** [comment]
Good point @youknowriad about WP 5.8 ❤️

**Comment by gziolo** [comment]
I applied changes myself and rebase the branch with `trunk` after I merged changes for the `Text` component in #31291.

I applied some additional changes:

- aligned the way we import components
- marked components as experimental in the documentation
- updates examples to use `__experimental` prefix

See https://github.com/WordPress/gutenberg/pull/31299/commits/8e21381239d824ad2a03a0b03ad88d67fdb57673.

This is how this component looks in Storybook:

Screen Shot 2021-04-30 at 12 06 41

**Review by gziolo — APPROVED** [review]

**Pull request merged**